### PR TITLE
two Crosswalk instances

### PR DESCRIPTION
**Pull request by dmwit**

### CODE REVIEW

File: Data/Align.hs [hunk 1]
```diff
@@ -368,6 +369,18 @@ crosswalkVector f = fmap VG.fromList . VG.foldr (alignWith cons . f) nil where
 instance Crosswalk V.Vector where
     crosswalk = crosswalkVector
 
+instance Crosswalk ((,) a) where
+    crosswalk fun (a, x) = fmap ((,) a) (fun x)
+
+-- can't (shouldn't) do longer tuples until there are Functor and Foldable
```
Comment (phadej) [inline Data/Align.hs]: aren't there? in `base-orphans` if not in `base`?

**Comment by dmwit** [comment]
Sure, I've added some tests. Didn't know about base-orphans, thanks for the pointer! It doesn't seem to define these instances, though -- it seems to be focused on providing instances that new bases have but old bases don't, and no base have Functor or Foldable instances for longer tuples. I know it's a weird situation, and I can't say I understand it, but there we are...

**Comment by phadej** [comment]
oh, dear, but then we have: http://hackage.haskell.org/package/base-4.11.1.0/docs/Data-Bifunctor.html

:disappointed:

**Comment by dmwit** [comment]
Oh, what, it just exports nothing if base is too old? That is a pretty surprising twist. Okay. I switched to base-compat-batteries before fully understanding your comment, but hang on just a tick and I'll remove that dependency in favor of just importing `Data.Functor.Compose`, since that seems cleaner overall, then do a force-push with a shorter commit list.

**Comment by phadej** [comment]
Released in https://hackage.haskell.org/package/these-0.7.5

**Pull request merged**